cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/realtime/kafka/Kafka08SimpleApiConsumer.java

```diff
@@ -402,6 +413,7 @@ private InetSocketAddress findLeader(String brokers, String topic, int partition
       } catch (Exception e) {
         LOG.error("Failed to communicate with broker {}:{} for leader lookup for topic-partition {}-{}",
                   broker.getKey(), broker.getValue(), topic, partition, e);
+        Throwables.propagate(e);
```

Comment:
what is the effect of this? Also, think its better for this method to throw the exception rather than propagate it as a runtime exception.